### DISCUSSION TITLE
Error with HMM analysis

### DISCUSSION THREAD

**Discussion by ffavelar** [p-j-smith/lipyphilic#117] [Q&A] [answered]
I'm trying to run the tutorial of the Hidden Markov Model in Ubuntu 20.04.6 LTS (GNU/Linux 5.15.0-75-generic x86_64) and Python 3.9.10. I get this output:

100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 51/51 [00:01<00:00, 29.28it/s]
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 51/51 [00:01<00:00, 46.18it/s]
  0%|                                                                                                                                                  | 0/3 [00:00<?, ?it/s]
MultinomialHMM has undergone major changes. The previous version was implementing a CategoricalHMM (a special case of MultinomialHMM). This new implementation follows the standard definition for a Multinomial distribution (e.g. as in https://en.wikipedia.org/wiki/Multinomial_distribution). See these issues for details:
https://github.com/hmmlearn/hmmlearn/issues/335
https://github.com/hmmlearn/hmmlearn/issues/340
  0%|                                                                                                                                                  | 0/3 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "HiddenMarkovModel.py", line 370, in <module>
    model = model.fit(
  File "/home/jorge/.local/lib/python3.9/site-packages/hmmlearn/base.py", line 469, in fit
    self._check()
  File "/home/jorge/.local/lib/python3.9/site-packages/hmmlearn/hmm.py", line 907, in _check
    raise ValueError(
ValueError: emissionprob_ must have shape (n_components, n_features)

I understand that emissionprob_ does not have the proper shape, but I don't know how to fix the code. Any help will be appreciated.

**Comment by p-j-smith** [comment, marked as answer]
Hi @ffavelar, thanks for raising this issue! It looks like `MultinomialHMM` has been renamed to `CategoricalHMM`, so in your script `HiddenMarkovModel.py` you will need to find the line:

```py
model = hmmlearn.hmm.MultinomialHMM(
        n_components=n_hidden_states,
        init_params='',
        verbose=True,
        n_iter=100  # maximum number of iterations
    )
```

and change it to:
```py
model = hmmlearn.hmm.CategoricalHMM(
        n_components=n_hidden_states,
        init_params='',
        verbose=True,
        n_iter=100  # maximum number of iterations
    )
```

Then hopefully everything should still work as before, but let me know if you have more issues

**Reply by ffavelar** [reply]
Hi, thank you for the reply. But now I get to the following error: 

Traceback (most recent call last):
  File "HiddenMarkovModel.py", line 290, in <module>
    emission_probs = get_emission_probs(
  File "HiddenMarkovModel.py", line 252, in get_emission_probs
    state_counts[state] = np.sum(states == state, axis=0)
ValueError: could not broadcast input array from shape (51,) into shape (204000,)

**Reply by p-j-smith** [reply]
so the issue is with this line in your script:
```py
state_counts[state] = np.sum(states == state, axis=0)
```
inside the function `get_emission_probs`. However, looking at the [tutorial notebook](https://github.com/p-j-smith/lipyphilic-tutorials/blob/main/notebooks/HiddenMarkovModel.ipynb) I can't see this line in this function - is it something you've added?

**Reply by ffavelar** [reply]
Sorry, I forgot that I messed a little with the code before contacting you. Now everything is working great. Thank you.

**Reply by p-j-smith** [reply]
No worries, glad it's working for you now 🙂
